scalameta/tokenizers/src/main/scala/scala/meta/internal/tokenizers/XmlParser.scala

```diff
@@ -0,0 +1,161 @@
+package scala.meta.internal.tokenizers
+
+/**
+  * Copy-pasta from this lihaoyi comment:
+  * https://github.com/scalameta/fastparse/pull/1#issuecomment-244940542*/
+import scala.annotation.switch
+
+import fastparse.all._
+
+class XmlParser(Block: P0,
+                Patterns: P0 = Fail,
+                WL: P0 = CharIn("\n ").rep(),
+                WS: P0 = " ".rep(),
+                HexNum: P0 = Fail) {
```

Comment:
Not sure if you want to solve this now, but these default values should really be something else.

Otherwise, XML literals including `CharRef`s will fail, XML literals with `\t`s inside will fail, and XML literal-pattern-matching will fail.

I'm sure you'll bump into this once you start trying to write unit tests and parsing legacy code, but if you can fix it up front maybe you should just do it. If not, kicking the can down the road also works.